src/rust/engine/BUILD

```diff
@@ -0,0 +1,36 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+files(
+    name="rust_sources",
+    sources=[
+        "Cargo.lock",
+        "build.rs",
+        "**/Cargo.toml",
+        "**/*.rs",
+        "**/*.proto",
+        ".cargo/config",
+        "rust-toolchain",
+    ],
+)
+
+shell_sources(name="shell")
+
+shell_command(
+    name="engine-and-client",
+    command=f"./cargo_build_shim.sh --features=extension-module -p engine -p client",
+    execution_dependencies=[
+        ":rust_sources",
+        "./cargo_build_shim.sh:shell",
+        "3rdparty/tools/python3:python3",
+    ],
+    extra_env_vars=["CHROOT={chroot}", "MODE"],
+    tools=["cc", "ld", "as", "ar", "cargo", "protoc", "python3.9"],
+    output_files=[
+        f"target/debug/libengine.so",
+        f"target/debug/pants",
+        f"target/release/libengine.so",
+        f"target/release/pants",
+    ],
+    timeout=600,
```

Comment:
So.... ummm........ Yeah. Not sure what to put :joy:

`1200` maybe?

Reply:
Probably `timeout=None`? Honestly, I'm surprised that `shell_command` has a default timeout... it probably shouldn't, or it should be much much longer.

Reply:
`None` just means the default 🫠